ports/nrf/common-hal/os/__init__.c

```diff
@@ -66,8 +67,25 @@ bool common_hal_os_urandom(uint8_t *buffer, uint32_t length) {
     uint8_t sd_en = 0;
     (void) sd_softdevice_is_enabled(&sd_en);
 
-    if (sd_en)
-        return NRF_SUCCESS == sd_rand_application_vector_get(buffer, length);
+    if (sd_en) {
+        while (length != 0) {
+            uint8_t available = 0;
+            sd_rand_application_bytes_available_get(&available);
+            if (available) {
+                uint32_t request = MIN(length, available);
+                uint32_t result = sd_rand_application_vector_get(buffer, request);
+                if (result != NRF_SUCCESS) {
+                    return false;
+                }
+                buffer += request;
+                length -= request;
+            } else {
+                RUN_BACKGROUND_TASKS;
+                tick_delay(500);
```

Comment:
Why delay?

Reply:
Doing it back to back usually causes no random bytes to be available yet.  In fact, usually even 1ms is too short for more random bytes to be available.  I think that in the softdevice, something interrupt-driven is happening, but not that frequently.  I can remove the delay, and it will just make the loop loop more times.

Reply:
removed.  It made no speed difference either way.